pyiceberg/manifest.py

```diff
@@ -308,6 +308,7 @@ def data_file_with_partition(partition_type: StructType, format_version: Literal
             field_id=field.field_id,
             name=field.name,
             field_type=partition_field_to_data_file_partition_field(field.field_type),
+            required=False
```

Comment:
Dafault value is True, which breaks the avro writer encoding manifestenrty when the partition value is null. (Because the attribute of required being True in the nested field of the avro schema will be used by the writer to infer and choose non-optional field writer which could not encode None value.)

This is for encoding only, so set it to False for now. 

TODO: Set it properly according to table schema.